### PR TITLE
CRM-20219 Update activity fields in profiles to use date picker

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
* [CRM-20219: Switch activity date time to use datepicker in profiles](https://issues.civicrm.org/jira/browse/CRM-20219)

### CODE REVIEW

File: CRM/Activity/Form/Task/Batch.php [hunk 1]
```diff
@@ -95,7 +95,7 @@ public function buildQuickForm() {
     $ufGroupId = $this->get('ufGroupId');
 
     if (!$ufGroupId) {
-      CRM_Core_Error::fatal('ufGroupId is missing');
+      throw new CRM_Core_Exception('The profile id is missing');
```
Comment (eileenmcnaughton) [inline CRM/Activity/Form/Task/Batch.php]: Just a general improvement, unrelated

File: CRM/Activity/Form/Task/Batch.php [hunk 2]
```diff
@@ -108,7 +108,7 @@ public function buildQuickForm() {
     $suppressFields = FALSE;
     $removehtmlTypes = array('File', 'Autocomplete-Select');
     foreach ($this->_fields as $name => $field) {
-      if ($cfID = CRM_Core_BAO_CustomField::getKeyID($name) &&
```
Comment (eileenmcnaughton) [inline CRM/Activity/Form/Task/Batch.php]: unused var, tidy up

File: CRM/Activity/Form/Task/Batch.php [hunk 3]
```diff
@@ -240,30 +239,9 @@ public function postProcess() {
...
-AND    ac.record_type_id = %2 )
-WHERE  a.id = %1 ";
-        $activityContacts = CRM_Core_OptionGroup::values('activity_contacts', FALSE, FALSE, FALSE, NULL, 'name');
-        $sourceID = CRM_Utils_Array::key('Activity Source', $activityContacts);
-        $params = array(1 => array($key, 'Integer'), 2 => array($sourceID, 'Integer'));
-        $dao = CRM_Core_DAO::executeQuery($query, $params);
-        $dao->fetch();
-
-        // Get Activity Type ID
-        $value['activity_type_id'] = $dao->activity_type_id;
```
Comment (eileenmcnaughton) [inline CRM/Activity/Form/Task/Batch.php]: all this just to backfill those fields if not in the form. Not required if we use the api

File: CRM/Activity/Form/Task/Batch.php [hunk 4]
```diff
@@ -228,10 +231,6 @@ public function postProcess() {
           $value['details'] = $value['activity_details'];
         }
 
-        if (!empty($value['activity_duration'])) {
```
Comment (eileenmcnaughton) [inline CRM/Activity/Form/Task/Batch.php]: tested & once using the api this doesn't add anything

**Comment by eileenmcnaughton** [comment]
@seamuslee001 I tried to figure out how to test this. In 4.6 there is a functionality to update an activity & contact at the same time, e.g at a url like 

civicrm/profile/edit?result=1&id={$contactID}&aid={$activityID}&gid={$profileID}

I am pretty sure that functionality is broken in 4.7, prior to this. 

The other place I could expose this field was batch  update from an activity search. I found some breakages pre-existing here (activity type is rendered as an integer), but I was able to demonstrate the activity date field updated ok

**Comment by seamuslee001** [comment]
@eileenmcnaughton  i tested the batch update and couldn't see any issue here so i think we can merge this

**Pull request merged**